451 abstract autofocus feature

annie-xd-wang opened the pull request.

annie-xd-wang committed:
update autofocus popup window #451

annie-xd-wang committed:
build/verify the autofocus parameters before launching the software #451

annie-xd-wang committed:
update autofocus popup controller #451

annie-xd-wang committed:
autofocus feature supports separate axis #451

annie-xd-wang committed:
add function to move remote focus device #451

annie-xd-wang committed:
autofocus feature support moving remote focus device #451

annie-xd-wang committed:
update tests

annie-xd-wang committed:
update autofocus parameter when loading a new experiment file

annie-xd-wang committed:
fix tests

zacsimile commented:
This is super cool and it seems to work! The only issue is we need to be able to pass a floating point range/step size in for remote_focus and the values need to be able to go negative.

![image](https://github.com/TheDeanLab/ASLM/assets/1263313/cfd97f06-d0cb-41da-86ea-b9003f81eca4)


It would also be nice if by default this launches with device `stage` and device reference `focus` by default.

AdvancedImagingUTSW committed:
Floating step size/range

Able to pass non-int values to the autofocus settings module. Works with synthetic mode. Needs to be validated.

It's not immediately clear from the presentation of the GUI if we are optimizing the remote focus amplitude or offset. If the amplitude, a negative value doesn't make sense. But I also do not necessarily see why the offset would need to go negative, since the autofocusing wizard is relative to the current value.

AdvancedImagingUTSW commented:
We should be able to test this out on CT-ASLM-V1.

AdvancedImagingUTSW committed:
Merge remote-tracking branch 'origin/develop' into 451-abstract-autofocus-feature

AdvancedImagingUTSW committed:
Fix merge error

Co-Authored-By: Zach Marin <zacsimile@users.noreply.github.com>

AdvancedImagingUTSW committed:
Fix test

Co-Authored-By: Zach Marin <zacsimile@users.noreply.github.com>

AdvancedImagingUTSW commented:
Confirmed working with changes in #503.
